media-sound/gstreamripper/files/gstreamripper-0.2-clang16-int-conversion.patch

```diff
@@ -0,0 +1,33 @@
+# Gentoo bug: https://bugs.gentoo.org/885531
+--- a/src/callbacks.c
++++ b/src/callbacks.c
+@@ -301,8 +301,8 @@ GIOChannel *set_up_io_channel(gint fd, GIOCondition cond, GIOFunc func,
+ 	GIOChannel *ioc;
+
+ 	/* set up handler for data */
+-	ioc = g_io_channel_unix_new(fd);
+-	pids[streams][1] = ioc;
++	ioc = (GIOChannel *)g_io_channel_unix_new(fd);
++	pids[streams][1] = (gint) ioc;
```

Comment:
Does it make sense to cast a GIOChannel* to a gint? https://docs.gtk.org/glib/ctor.IOChannel.unix_new.html says it returns a GIOChannel, which is what you'd expect.

What is `pids` and what is it supposed to contain?

I don't see a function at https://docs.gtk.org/glib/struct.IOChannel.html to get out an FD or something though...

Reply:
pids are gint. What's it supposed to contain I'm not sure. But guessing from the name it could be process IDs?

Reply:
Instead of casting should I make `pids` GIOChannel?